src/shared/components/axisTicks/TimeTickInput.tsx

```diff
@@ -0,0 +1,236 @@
+    setOptions(tickOptionNameWithoutAxis, convertedValue)
+    if (convertedValue === convertedValue) {
+      setOptions('GenerateAxisTicks', [
+        ...filteredTickOptions,
+        tickOptionNameWithAxis,
+      ])
+    } else {
+      setOptions('GenerateAxisTicks', filteredTickOptions)
+      if (tickOptionInput !== '') {
+        setTickOptionInputStatus(ComponentStatus.Error)
+      }
+    }
+  }
+
+  const handleInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setTickOptionInput(event.target.value)
+    setTickOptionInputStatus(ComponentStatus.Default)
+  }
+
+  const handleBlur = () => updateTickOption()
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      updateTickOption()
+    }
+  }
+
+  const getDatePickerDateTime = () => {
+    const date = new Date(tickOptionInput)
+    if (date.valueOf() === date.valueOf()) {
+      return date.toISOString()
+    }
+    return new Date().toISOString()
+  }
+
+  const handleReset = () => {
+    setTickOptionInput('')
+    updateTickOption('')
+  }
+
+  const showDatePicker = () => setIsDatePickerOpen(true)
+  const hideDatePicker = () => setIsDatePickerOpen(false)
+
+  const toggleDatePicker = () => {
+    if (isDatePickerOpen) {
+      hideDatePicker()
+    } else {
+      showDatePicker()
+    }
+  }
+
+  const handleSelectDate = (date: string) => {
+    const dateRFC3339 = new Date(date).valueOf()
+    setTickOptionInput(dateRFC3339 === dateRFC3339 ? String(dateRFC3339) : '')
+    setTickOptionInputStatus(ComponentStatus.Default)
+    updateTickOption(dateRFC3339)
+  }
+
+  const onClickOutside = () => {
+    if (isOnClickOutsideHandlerActive) {
+      hideDatePicker()
+    }
+  }
+
+  const allowOnClickOutside = () => setIsOnClickOutsideHandlerActive(true)
+  const suppressOnClickOutside = () => setIsOnClickOutsideHandlerActive(false)
+
+  const style: CSSProperties = isDatePickerOpen
+    ? {position: 'relative'}
+    : {
+        top: `${window.innerHeight / 2}px`,
+        left: `${window.innerWidth / 2}px`,
+        transform: `translate(-50%, -50%)`,
+      }
+
+  return (
+    <>
+      <Grid.Column widthXS={Columns.Six}>
+        <Form.Element label="Tick Start">
+          <Input
+            placeholder={placeholder}
+            onChange={handleInput}
+            onFocus={handleInput}
+            onBlur={handleBlur}
+            onKeyPress={handleKeyPress}
+            value={tickOptionInput}
+            status={tickOptionInputStatus}
+          />
+        </Form.Element>
+      </Grid.Column>
+      <Grid.Column widthXS={Columns.Six}>
+        <Form.Element label="Date Picker">
+          <Popover
+            appearance={Appearance.Outline}
+            position={PopoverPosition.ToTheLeft}
+            triggerRef={triggerRef}
+            visible={isDatePickerOpen}
+            showEvent={PopoverInteraction.None}
+            hideEvent={PopoverInteraction.None}
+            distanceFromTrigger={8}
```

Comment:
magic number! what's 8 mean?

Reply:
According to clockface, this is the pixel distance between the trigger (our button) and the popover's dialog box.

All of the `Popover`s in ui have `8` as the value. I just copied it. Perhaps we can clean that up all in one pass rather than piecemeal. Can we leave this for another PR?

Reply:
>Can we leave this for another PR?

sure, sounds like a good idea